Docs: Add bz2 usage examples

bsolomon1124 opened the pull request:
- Adds an "Examples of usage" section inspired by the one
  found in the gzip docs
- Corrects the descriptions for ``compresslevel`` and ``data``:
    - ``compresslevel`` must be an `int`, not any number.  For
      instance, passing a float will raise ``TypeError``
    - Note that `data` must be bytes-like

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->

pitrou commented on Doc/library/bz2.rst:
```diff
@@ -254,3 +254,75 @@ One-shot (de)compression
+Examples of usage
+-----------------
+
+Below are some examples of typical usage of the :mod:`bz2` module:.
```
```suggestion
Below are some examples of typical usage of the :mod:`bz2` module.
```

pitrou reviewed (commented):
LGTM generally. I'll let someone else review the grammar.

bsolomon1124 committed:
Docs: Add bz2 usage examples

- Adds an "Examples of usage" section inspired by the one
  found in the gzip docs
- Corrects the descriptions for ``compresslevel`` and ``data``:
    - ``compresslevel`` must be an `int`, not any number.  For
      instance, passing a float will raise ``TypeError``
    - Notes that `data` must be bytes-like

gpshead committed:
remove wrongly placed 502 return value

gpshead committed:
fix for the previous doctest "fix" :)

gpshead committed:
call the unused value "unused" to avoid confusion.

gpshead committed:
remove trailing whitespace.

gpshead committed:
remove `` from within docstring in example code.

bsolomon1124 committed:
Remove trailing space from bytestring

This modifies the compression ratio by a hair.

bsolomon1124 committed:
Use 'nonrandom' data in compression example

Compressing random data (low compression ratio) is a
less realistic and less useful example.

bsolomon1124 committed:
Remove empty prompt

Per https://devguide.python.org/documenting/.
